http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/ChannelPipelineInitializer.java

```diff
@@ -94,6 +96,8 @@ public void channelCreated(Channel ch) {
 
             pipeline.addLast(sslHandler);
             pipeline.addLast(SslCloseCompletionEventHandler.getInstance());
+
+            ch.config().setOption(ChannelOption.ALLOCATOR, UnpooledByteBufAllocator.DEFAULT);
```

Comment:
Should we only set this for JDK `SsLProvider` since the issue only affects customers using JDK SSL engine?

Reply:
@dagnir @millems WDUT?

Reply:
+1

Reply:
+1, smallest blast-radius is better. Easier to expand later if it causes problems.

Reply:
Got it, will do!